runtime/gc_realtime/RealtimeGC.cpp

```diff
@@ -806,51 +303,8 @@ MM_RealtimeGC::incrementalCollect(MM_EnvironmentRealtime *env)
 	_sched->run(env, &markTask);
 	reportMarkEnd(env);
 	
-	/* 
-	 * We have to do class unloading before we turn on GC_UNMARK allocation collor.
-	 */
 
-#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
-	if (_extensions->runtimeCheckDynamicClassUnloading != 0) {
-		MM_ClassUnloadStats *classUnloadStats = &_extensions->globalGCStats.classUnloadStats;
-		setCollectorUnloadingClassLoaders();
-		reportClassUnloadingStart(env);
-		classUnloadStats->_startTime = j9time_hires_clock();
-		unloadDeadClassLoaders(env);
-		classUnloadStats->_endTime = j9time_hires_clock();
-		reportClassUnloadingEnd(env);
-		
-		/* If there was dynamic class unloading checks during the run, record the new number of class
-		 * loaders last seen during a DCU pass
-		 */
-		_extensions->classLoaderManager->setLastUnloadNumOfClassLoaders();		
-		_extensions->classLoaderManager->setLastUnloadNumOfAnonymousClasses();
-	}
-	
-	/* Handling of classes done. Return back to "mark if necessary" mode */
-	_unmarkedImpliesClasses = false;
-
-	/* Clear the appropriate flags of all classLoaders */
-	GC_ClassLoaderIterator classLoaderIterator(_javaVM->classLoaderBlocks);
-	J9ClassLoader *classLoader;
-	while((classLoader = classLoaderIterator.nextSlot()) != NULL) {
-		classLoader->gcFlags &= ~J9_GC_CLASS_LOADER_SCANNED;
-	}
-#endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
-	
-	/* If the J9VM_DEBUG_ATTRIBUTE_ALLOW_USER_HEAP_WALK flag is set,
-	 * or if we are about to unload classes and free class memory segments
-	 * then fix the heap so that it can be walked by debugging tools
-	 */
-#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
-	bool fixupForClassUnload = (_extensions->classLoaderManager->reclaimableMemory() > _extensions->deadClassLoaderCacheSize);
-#else /* defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING) */
-	bool fixupForClassUnload = false;
-#endif /* defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING) */
-	if (J9VM_DEBUG_ATTRIBUTE_ALLOW_USER_HEAP_WALK == (((J9JavaVM *)env->getLanguageVM())->requiredDebugAttributes & J9VM_DEBUG_ATTRIBUTE_ALLOW_USER_HEAP_WALK)
-			|| fixupForClassUnload) {
-		_fixHeapForWalk = true;
-	}	
+	_realtimeDelegate.incrementalCollect(env);
```

Comment:
`_dynamicClassUnloadingEnabled ` use to be set before Mark. Currently however it has been moved to Delegate and set after Mark